### DISCUSSION TITLE
Turning on charging from Web UI

### DISCUSSION THREAD

**Discussion by devdems** [dingo35/SmartEVSE-3.5#46] [Q&A] [answered]
Not sure if i'm doing something wrong but I turned off charging by clicking on OFF from the WEB UI. I have tried to turn it on by clicking SMART, SOLAR, and NORMAL but none of those options worked. 

I had to go physically to the SmartEVSE and hold the button for 2s so it started. 

I also tried to change mode through the API but it did not change it.

Any ideas?

**Comment by dingo35** [comment]
You sure you didnt activate the timer?

EDIT: spiffs.bin properly flashed?
EDIT2: check if your meters are still properly configured; see release notes v3.5.0-RC2

**Comment by devdems** [comment]
@dingo35 jup, the upgrade was done correctly and the meter is also properly configured and working as I can see all the data correctly from it.

All I did was open the webpage and click the SMART button.

**Reply by dingo35** [reply]
I cant reproduce it, can you?

Perhaps refreshing your webpage or emptying the cache of your browser?

IMHO nothing changed in that part of the code....

**Comment by devdems** [comment]
@dingo35 yeah, i can reproduce it over and over again.
Step 1: Open webpage and click on OFF button. OFF button turns green and mode is off.
Step 2: Click on SMART button. Nothing happens. Mode is still off and I can't charge it unless I go physically to the device and hold the button for 2 seconds to work normally again.

![image](https://github.com/dingo35/SmartEVSE-3.5/assets/8215353/e078f76f-b77f-4851-9656-54939a6d772b)

I have tested it on different browsers and also in incognito mode.

**Reply by devdems** [reply]
One more thing. This also could maybe not work from the beginning because I have never used this before. So it does not mean that it would come from RC2.

**Reply by dingo35** [reply]
Ok that is important info, thx for sharing.

When I look at your screenshot your time has not synced, so it looks like your wifi signal might be unstable?

**Reply by dingo35** [reply]
Also, what time zone are you in? It looks like you are using US localisation settings ?

**Comment by dingo35** [comment]
[88d9d5c18.zip](https://github.com/dingo35/SmartEVSE-3.5/files/14908999/88d9d5c18.zip)

This is the current version, post-RC2. Does it have the problem too? If so, please post your /settings  ....

**Reply by devdems** [reply]
Ok, so i have made a successful update with two OK's.

Mode was off. I still can't change it to smart, normal or solar.

On the device it's written: "STARTING @ Sun 0 Jan"

When holding the button for 2s it's working normally.

I'm in CEST timezone. Not sure what you mean regarding localisation setting. I should be using Slovene but i do not see any option to change that on the device so it's using the default one probably?


`{"version":"17:21:46 @Apr  8 2024","mode":"OFF","mode_id":0,"car_connected":false,"wifi":{"status":"WL_CONNECTED","ssid":"MATRIX","rssi":-60,"bssid":"48:A9:8A:34:E4:09"},"evse":{"temp":32,"temp_max":65,"connected":false,"access":false,"mode":2,"loadbl":0,"pwm":1024,"solar_stop_timer":0,"state":"Ready to Charge","state_id":0,"error":"None","error_id":0,"rfid":"Not Installed"},"settings":{"charge_current":0,"override_current":0,"current_min":6,"current_max":20,"current_main":25,"current_max_circuit":20,"current_max_sum_mains":60,"solar_max_import":0,"solar_start_current":4,"solar_stop_time":10,"enable_C2":"Not present","modem":"Not present","mains_meter":"SolarEdge","starttime":1712599380,"stoptime":0,"repeat":0},"mqtt":{"host":"10.0.0.10","port":1883,"topic_prefix":"SmartEVSE-45332","username":"","password_set":false,"status":"Connected"},"home_battery":{"current":0,"last_update":0},"ev_meter":{"description":"Disabled","address":12,"import_active_power":0,"total_kwh":0,"charged_kwh":0,"currents":{"TOTAL":0,"L1":0,"L2":0,"L3":0},"import_active_energy":0,"export_active_energy":0},"mains_meter":{"import_active_energy":18601.40039,"export_active_energy":17866.80078},"phase_currents":{"TOTAL":60,"L1":30,"L2":23,"L3":7,"last_data_update":2785657,"original_data":{"TOTAL":60,"L1":30,"L2":23,"L3":7}},"backlight":{"timer":0,"status":"OFF"}}`

**Comment by dingo35** [comment]
Alternatively, you could test to downgrade to serkri v1.8.0, that version has no webserver problems 100% sure...

**Reply by devdems** [reply]
I've downgraded to serki but the same problem exists so it's not connected to your changes in RC versions. I also tried updating it with API which also does not work. The request goes trough without an error but when you check it the mode did not change.

![image](https://github.com/dingo35/SmartEVSE-3.5/assets/8215353/0c652c1b-5920-47ed-9d2f-b1dc86766e74)

![image](https://github.com/dingo35/SmartEVSE-3.5/assets/8215353/75e27b93-6613-4ffa-8052-9a30add45b27)

**Comment by dingo35** [comment]
The message "STARTING" means you are activating delayed charging timer.

Please read the full manual, especially "operation" .

Also, after reboot system time should set itself via ntp; SmartEVSE looks for local ntp server on your LAN and if not succesful it looks on the internet.

If after 60s after reboot you still dont have a normal system time, your LAN is blocking internet access to ntp server and jquery.

**Comment by dingo35** [comment]
Also you maxsummains should be set at the default of 600A.

**Comment by devdems** [comment]
Ok, i see what was the problem. SmartEVSE needs NTP that the router needs to provide. I did not have it set on my Mikrotik routers because I never needed this. Now when I have added it to the router it has the correct time and I can switch through the modes.

Not sure this is the best approach that we are taking here. Maybe we should give a user option to input public NTP server or at least mention this as mandatory in the instructions.

**Comment by dingo35** [comment]
If your local router does  not provide ntp it will automatically sync with a public ntp server. 
Your mikrotik probably blocks ntp comms, or your DNS has problems.

**Reply by devdems** [reply]
Currently, i don't see that network is blocking anything, so this is still treated as a bug. Will check more thoroughly tomorrow.

**Comment by dingo35** [comment]
Too bad you are using up dev time for a LAN config problem...

**Reply by devdems** [reply]
I don't agree. This was opened in the discussions section.

**Comment by dingo35** [comment, marked as answer]
Apologies, turns out the underlying libraries blocked the use of a fallback ntp server since somewhere in 2021; nobody noticed because almost all home network routers have local ntp server enabled, except (apparently) Mikrotik.

I now have to block the use of the local ntp server in favor of using public ntp server...
See 5707a8f5f0

**Reply by devdems** [reply]
Interesting. Can't we do an if here so we can keep the fallback function?

**Reply by devdems** [reply]
Scratch that. I see from the link that you have provided that this is not such an easy fix. Choosing the public ntp is the best approach.
